tsdb/wal/wal_test.go

```diff
@@ -484,3 +484,32 @@ func BenchmarkWAL_Log(b *testing.B) {
 		})
 	}
 }
+
+// TestCloseGoroutineLeak detect goroutine leak when call Close of wal instance.
+func TestCloseGoroutineLeak(t *testing.T) {
+	dir, err := ioutil.TempDir("", "wal_close_goroutine_leak")
+	testutil.Ok(t, err)
+	defer func() {
+		testutil.Ok(t, os.RemoveAll(dir))
+	}()
+	w, err := NewSize(nil, nil, dir, pageSize, false)
+	testutil.Ok(t, err)
+
+	closedSegment, err := CreateSegment(dir, 0)
```

Comment:
Because it's not a closed segment when you create it.
```suggestion
	segment, err := CreateSegment(dir, 0)
```